software/contrib/poly_square.py

```diff
@@ -0,0 +1,246 @@
+        # Settings for improved performance
+        machine.freq(250_000_000)
+        k1.set_samples(32)
+        k2.set_samples(32)        
+        # PIO settings
+        max_count = 1_000_000
+        count_freq = 50_000_000
+        self.oscillators = [
+            SquareOscillator(0, 21, max_count, count_freq),
+            SquareOscillator(1, 20, max_count, count_freq),
+            SquareOscillator(2, 16, max_count, count_freq)
+        ]
+        # To add more polyphony modes, include them in this list. The offsets
+        # are V/oct offsets (ie, a 5th = 7/12, an octave = 1, etc.). The number
+        # of offsets in the tuple must match the length of the self.oscillators
+        # list above.
+        self.modes = [
+            PolyphonyMode("Unison", (0, 0, 0)),
+            PolyphonyMode("5th", (0, 0, 7/12)),
+            PolyphonyMode("Octave", (0, 0, 1)),
+            PolyphonyMode("Power chord", (0, 7/12, 1)),
+            PolyphonyMode("Stacked 5ths", (0, 7/12, 14/12)),
+            PolyphonyMode("Minor triad", (0, 7/12, 15/12)),
+            PolyphonyMode("Major triad", (0, 7/12, 16/12)),
+            PolyphonyMode("Diminished", (0, 6/12, 15/12)),
+            PolyphonyMode("Augmented", (0, 8/12, 16/12)),
+            PolyphonyMode("Major 6th", (0, 4/12, 9/12)),
+            PolyphonyMode("Major 7th", (0, 4/12, 11/12)),
+            PolyphonyMode("Minor 7th", (0, 3/12, 10/12))
+        ]
+        self.current_mode = None
+        self.ui_update_requested = True
+        self.detune_amount = None
+        self.coarse_tune = .5
+        self.fine_tune = .5
+        self.tuning_mode = False
+        self.load_state()
+
+        @b1.handler
+        def tuning_mode_on():
+            self.tuning_mode = True
+            self.ui_update_requested = True
+
+        @b1.handler_falling
+        def tuning_mode_off():
+            self.tuning_mode = False
+            self.ui_update_requested = True
+            # Saves the tuning settings when the button is released
+            self.save_state()
+        
+    # Converts V/oct signal to Hertz, with 0V = C0
+    def get_hertz(self, voltage):
+        # Start with A0 because it's a nice, rational number
+        a0 = 27.5
+        # Subtract 3/4 from the voltage value so that 0V = C0
+        return a0 * 2 ** (voltage - 3/4)
+    
+    # Returns the linear step distance between elements such that all are
+    # equally spaced. Ex, for 6 elements between 0 and 100 (inclusive), the
+    # step would be 20 (the elements would then be 0, 20, 40, 60, 80, 100).
+    def get_step_distance(self, first, last, count):
+        return (last - first) / (count - 1)
+    
+    # Returns the total voltage offset of the current tuning values
+    def get_tuning(self):
+        return self.coarse_tune - .5 + (self.fine_tune - .5) / 12
+    
+    # Through-zero detuning of current oscillator based on its position in
+    # the list of oscillators - the central oscillator (assuming an odd 
```

Comment:
Yes, it is currently set up as oscillator 2 - detuning basically sets a maximum positive and negative amount and then spreads the voices across that, so that's why the first oscillator is tuned flat. I guess this could be adjusted easily by changing the pin that each oscillator is connected to.